app/models/user.rb

```diff
@@ -63,9 +63,8 @@ def previous_upload_for(category)
   end
 
   def pbs
-    runs.where(
-      id: categories.map { |category| pb_for(category).id10 } | runs.where(category: nil).pluck(:id)
-    )
+    Run.where(user: self).select('DISTINCT ON (category_id) *').order('category_id, realtime_duration_ms ASC')
+       .union(Run.where(user: self, category: nil))
```

Comment:
Same question as above, also there's the `runs.by_category(nil)` scope available if that is equivalent to this